CONTRIBUTING.md

```diff
@@ -97,33 +104,114 @@ Categories of system props are exported from `src/constants.js`:
+
+```
+yarn test -- --watch
 ```
 
-This will create a new deployment and alias it to its production URL, [primer-components.now.sh](https://primer-components.now.sh).
+See [`src/__tests__/example.js`](src/__tests__/example.js) for examples of ways that we test our components.
+
+### TypeScript support
+
+Several of the projects that consume Primer Components are written in TypeScript! Though Primer Components is not currently written in TS, we do export type definitions in order to make Primer Components compatible with other TS projects.
+
+Whenever adding new components or modifying the props of an existing component, **please make sure to update the type definition** in `index.d.ts`! This is super important to make sure we don't break compatibility :) 
+
+### Additional resources
+
+- [Primer Components Philosophy](https://primer.style/components/philosophy)
+- [Primer Components Core Concepts](https://primer.style/components/core-concepts)
+- [Primer Components System Props](https://primer.style/components/system-props)
+- [Styled Components docs](https://styled-components.com/)
+- [Styled System docs](https://styled-system.com/)
+
+## Writing documentation
+
+We use [Doctocat](https://github.com/primer/doctocat) to power our documentation site at [https://primer.style/components](https://primer.style/components/).
+
+To add a new component to our documentation site, create a new file with the `.md` extension for your component in `docs/content/components`.
+
+## Creating a pull request
+
+When you're ready to create a pull request, open one using the pull request template. Be sure to add screenshots of any relevant work and a thoughtful description.
+
+### What to expect after opening a pull request
+After opening a pull request, a member of the design systems team will add the appropriate labels (major, minor, patch release labels) and update the base branch to the correct release branch. Usually, you'll receive a response from the design systems team within a day or two. The design systems team member will review the pull request keeping the following items in mind:
+
+
+### What we look for in reviews
+- If it's a new component, does the component make sense to add to Primer Components? (Ideally this is discussed before the pull request stage, please reach out to a DS member if you aren't sure if a component should be added to Primer Componnets!)
+- Does the component follow our Primer Components code style?
+- Does the component use theme values for most CSS values?
+- Does the component have access to the [default theme](#adding-default-theme)?
+- Does the component have the [correct system props implemented](#adding-system-props)?
+- Is the component API intuitive?
+- Does the component have the appropriate [type definitions in `index.d.ts`](#typescript-support)?
+- Is the component documented accurately?
+- Does the component have appropriate tests?
+- Does the pull request increase the bundle size significantly?
```

Comment:
I'd love to have some automated way to enforce a small bundle size. @siddharthkp's bundlesize project looks promising: https://github.com/siddharthkp/bundlesize

Reply:
Yes! I was thinking that too

Reply:
I was thinking of that too 😄